### PR TITLE
Handled user account failure and sender acknowledgement

### PR DESCRIPTION
**Pull request by Edzelopez**

### CODE REVIEW

File: aoservicelisting.civix.php [hunk 1]
```diff
@@ -307,15 +308,7 @@ public static function emailRule($cid) {
...
-      'contact_id_b' => $cid,
-      'relationship_type_id' => PRIMARY_CONTACT_REL,
-      'return' => 'contact_id_a',
-    ]);
-    if ($relationship['count'] > 0 && !empty($relationship['values'][$relationship['id']]['contact_id_a'])) {
-      $cid = $relationship['values'][$relationship['id']]['contact_id_a'];
-    }
-    else {
+    if (empty($cid)) {
       return;
```
Comment (seamuslee001) [inline aoservicelisting.civix.php]: should we return FALSE here?

Comment (Edzelopez) [reply aoservicelisting.civix.php]: Yep, updated.

File: aoservicelisting.civix.php [hunk 2]
```diff
@@ -353,11 +347,24 @@ function setStatus($oldStatus = NULL, $cid, $submitValues) {
...
+        ]);
+        if ($relationship['count'] > 0 && !empty($relationship['values'][$relationship['id']]['contact_id_a'])) {
+          $primaryCid = $relationship['values'][$relationship['id']]['contact_id_a'];
+        }
         // Create drupal account if not exists.
-        self::createUserAccount($cid);
+        if (!self::createUserAccount($primaryCid)) {
+          // Set status message indicating that user account creation was unsuccessful.
+          $userUrl = CRM_Utils_System::url('civicrm/contact/view/useradd', "reset=1&action=add&cid=$primaryCid");
+          CRM_Core_Session::setStatus(ts('There was an error creating the user account. Please proceed to add one here: %1', [1 => $userUrl]), ts('Warning'), 'alert');
```
Comment (seamuslee001) [inline aoservicelisting.civix.php]: @Edzelopez I tested this  by altering the primary contact for one of my test records to include an `=` in the last name. I can confirm that no user account was created, However no pop box showed indicating there was an error creating the user.

File: aoservicelisting.civix.php [hunk 3]
```diff
@@ -353,11 +347,24 @@ function setStatus($oldStatus = NULL, $cid, $submitValues) {
...
+        }
 
         // Send Mail
-        self::sendMessage($cid, APPROVED_MESSAGE);
+        self::sendMessage($primaryCid, APPROVED_MESSAGE);
```
Comment (seamuslee001) [inline aoservicelisting.civix.php:372]: Can confirm that this change causes the approval email to go to the primary contact now correctly

**Comment by seamuslee001** [comment]
Changes look good to me

**Review by seamuslee001 — APPROVED** [review]

**Pull request merged**